### PR TITLE
TRUNK-5799: EncounterProvider Domain - Switching from Hibernate Mappings to Annotations

### PR DESCRIPTION
**Pull request by alinmihaila99**
## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5799

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by alinmihaila99** [comment]
Hi @Ruhanga, can you review this?

File: api/src/test/java/org/openmrs/api/OrderServiceTest.java [hunk 1]
```diff
@@ -2744,7 +2745,8 @@ public void saveOrder_shouldFailIfTheJavaTypeOfThePreviousOrderDoesNotMatch() th
 		Metadata metaData = new MetadataSources(standardRegistry).addAnnotatedClass(Allergy.class)
 				.addAnnotatedClass(Encounter.class).addAnnotatedClass(SomeTestOrder.class)
 				.addAnnotatedClass(Diagnosis.class).addAnnotatedClass(Condition.class)
-				.addAnnotatedClass(Visit.class).getMetadataBuilder().build();
```
Comment (gitcliff) [inline api/src/test/java/org/openmrs/api/OrderServiceTest.java:2747]: @alinmihaila99 what could be the reason behind this change?

Comment (HerbertYiga) [reply api/src/test/java/org/openmrs/api/OrderServiceTest.java:2747]: @gitcliff  that change was necessary to make sure that EncounterProvider is also added to the list of the annotated classes,checkout the code snipet below that

File: api/src/main/java/org/openmrs/EncounterProvider.java [hunk 2]
```diff
@@ -9,21 +9,44 @@
...
+@BatchSize(size = 25)
 public class EncounterProvider extends BaseChangeableOpenmrsData {
-	
+
 	public static final long serialVersionUID = 1L;
-	
+
+	@Id
+	@Column(name = "encounter_provider_id")
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
```
Comment (Ruhanga) [inline api/src/main/java/org/openmrs/EncounterProvider.java:37]: @alinmihaila99 do you need to use the `GeneratedValue` annotation here?

**Review by Ruhanga — CHANGES_REQUESTED** [review]
@alinmihaila99, can you resolve the merge conflicts here?

**Review by Ruhanga — APPROVED** [review]

**Pull request closed without merging**